src/Widgets/ImageWidget/DimensionsEditor.tsx

```diff
@@ -59,9 +59,12 @@ const DimensionEditor = connect(function DimensionEditor({
       <div className="item_content">
         <div className="input_group" aria-readonly={readOnly}>
           <input
+            max={unit === '%' ? 100 : undefined}
+            min={0}
             onChange={({ target: { value } }) => updateValue(value)}
             placeholder={readOnly ? undefined : '100'}
             readOnly={readOnly}
+            step="any"
```

Comment:
FMI: Why is the `step` attribute needed?

Reply:
Without `step`, using up/down would round to the next value with the default step granularity `1`. E.g., If the value is `12.5`, pressing "up" will result in `13` instead of `13.5`. Step `any` keeps the fractional digits. This is more user-friendly as it prevents accidental loss of data. It's also more in line with, e.g., what the Chrome dev tools do with CSS values up/down.